src/main/java/seedu/address/model/Statistics.java

```diff
@@ -23,33 +28,120 @@ public Statistics(ReadOnlyStatistics toBeCopied) {
         resetData(toBeCopied);
     }
 
+    /** Initialises empty DayData for past MAX_SIZE days */
+    public void init() {
+        LocalDate currDate = LocalDate.now();
+        for (int i = 0; i < MAX_SIZE; i++) {
+            LocalDate tempLocalDate = currDate.minusDays(i);
+            String tempLocalDateStr = tempLocalDate.toString();
+            Date tempDate = new Date(tempLocalDateStr);
+            DayData dayData = new DayData(tempDate);
+            dayDataList.add(dayData);
+        }
+    }
+
     //// list overwrite operations
-    /**
-     * Replaces the contents of the person list with {@code persons}. {@code persons} must not
-     * contain duplicate persons.
-     */
+
+    /** Clears list */
+    public void clearList() {
+        this.dayDataList.clear();
+    }
+
+    /** Replaces the contents of the list with {@code dayDataList}. */
     public void setDayDatas(List<DayData> dayDataList) {
         this.dayDataList.clear();
         this.dayDataList.addAll(dayDataList);
     }
 
-    /** Resets the existing data of this {@code TaskList} with {@code newData}. */
+    /** Resets the existing data of this {@code Statistics} with {@code newData}. */
     public void resetData(ReadOnlyStatistics newData) {
         requireNonNull(newData);
         setDayDatas(newData.getDayDataList());
     }
 
-    //// person-level operations
+    //// dayData-level operations
+
+    // FUNCTIONS FOR
+    // HARDOHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH SIR
+
+    /** reinitialises dayDataList to current day while retaining stored data. */
+    public void fetch() {
+        HashMap<Date, PomDurationData> pomTempStorage = new HashMap<>();
+        HashMap<Date, TasksDoneData> tasksTempStorage = new HashMap<>();
+
+        for (int i = 0; i < dayDataList.size(); i++) {
+            pomTempStorage.put(
+                    dayDataList.get(i).getDate(), dayDataList.get(i).getPomDurationData());
+            tasksTempStorage.put(
+                    dayDataList.get(i).getDate(), dayDataList.get(i).getTasksDoneData());
+        }
+
+        this.init();
+
+        for (int i = 0; i < dayDataList.size(); i++) { // updated dayDatalist
+            Date currTempDate = dayDataList.get(i).getDate();
+            if (pomTempStorage.containsKey(currTempDate)) {
+                PomDurationData tempPomDurationDate = pomTempStorage.get(currTempDate);
+                TasksDoneData tempTasksDoneData = tasksTempStorage.get(currTempDate);
+                DayData tempDayData =
+                        new DayData(currTempDate, tempPomDurationDate, tempTasksDoneData);
+                dayDataList.remove(i);
+                dayDataList.add(i, tempDayData);
+            }
+        }
+
+        assert (dayDataList.size() <= MAX_SIZE);
+    }
+
+    /**
+     * Auto-checks and replaces a new DayData object at the same Date
+     *
+     * @param dayData
+     */
+    public void updatesDayData(DayData dayData) {
```

Comment:
This should only be done on today's data? When can a user go back and edit yesterday's stats?

Reply:
This function is not supposed to be accessed by users (enforced in modelmanager test). But there is a need for developer to update yesterday's stats when pomodoro timer crosses to the next day